packs/email/sensors/imap_sensor.py

```diff
@@ -41,17 +41,18 @@ def __init__(self, sensor_service, config=None, poll_interval=30):
     def setup(self):
         self._logger.debug('[IMAPSensor]: entering setup')
 
-        if 'imap_mailboxes' in self._config:
-            self._parse_mailboxes(self._config['imap_mailboxes'])
-
     def poll(self):
         self._logger.debug('[IMAPSensor]: entering poll')
 
+        if 'imap_mailboxes' in self._config:
+            self._parse_mailboxes(self._config['imap_mailboxes'])
```

Comment:
Ok! After some :tea:, I am human again.

I put some details on this on commit https://github.com/StackStorm/st2contrib/commit/a5d6e76e6e60616ed5d51b46895cbe89d5bf5a4a. I was having to do deep traces into the Python IMAP library and how it implemented RFCs as I was getting protocol errors. Time to fix vs overhead, I went this way to save some time. If we need to optimize for a long poll, we'll have to ditch the `easyimap` library and get closer to the library and protocol.